site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -0,0 +1,290 @@
+
+    constructor(props) {
+        super(props);
+        this.FormSchema = Yup.object({
+            name: Yup.string().required(),
+            email: Yup.string().email().required(),
+            category: Yup.string().required(),
+            subject: Yup.string().required(),
+            description: Yup.string().min(
+                this.props.descriptionMinLength,
+                `Field must be at least ${this.props.descriptionMinLength} characters long.`
+            ).max(
+                this.props.descriptionMaxLength,
+                `Field must be at most ${this.props.descriptionMaxLength} characters long.`)
+                .required(),
+            sysInfo: Yup.boolean()
+        });
+
+        this.state = {
+            files: []
+        };
+    }
+
+    onSubmit = async (values, formikBag) => {
+        const { apiEndpoint } = this.props;
+        let response;
+        const formData = formikToFormData(values);
+        try {
+            response = await http.post(apiEndpoint, formData);
+        } catch (error) {
+            let errMessage = error.message;
+            if (response) {
+                errMessage = error.response.data;
+            }
+            
+            // TODO show error notification
+            console.error(errMessage); 
+        }
+
+        // TODO show a success notification
+        window.location = response.request.responseURL;
+    }
+
+    updateFiles = (files, formikSetFieldValue) => {
+        formikSetFieldValue("files", files)
+        this.setState({
+            files: files
+        });
+    }
+
+    render() {
+        const {
+            categories,
+            name,
+            isUserAuthenticated,
+            userBrowser,
+            userPlatform,
+            userMail,
+            maxFileSize
+        } = this.props;
+
+        const defaultCategory = categories.length > 0 ? categories[0].key : '';
+
+        const initialValues = {
+            email: userMail,
+            name: name,
+            category: defaultCategory,
+            sysInfo: false,
+            files: null
+        }
+
+        const sysInfo = `Browser: ${userBrowser} Operating System: ${userPlatform}`;
+
+        return (
+            <Formik
+                initialValues={initialValues}
+                onSubmit={this.onSubmit}
+                validationSchema={this.FormSchema}
+                validateOnChange={false}
+                validateOnBlur={false}
+            >
+                {({ handleSubmit, setFieldValue }) => {
+                    let dropzoneParams = {
+                        onDropAccepted: (acceptedFiles) => {
+                            this.updateFiles(acceptedFiles, setFieldValue)
+                        },
+                        onDropRejected: (rejectedFiles) => {
+                            // TODO: show error message when files are rejected e.g size limit
+                            console.error(rejectedFiles[0].errors);
```

Comment:
Needs to be implemented: when the file is rejected by `react-dropzone`, we should somehow notify the user.

Reply:
Working on https://github.com/zenodo/zenodo-rdm/issues/84